internal/component/loki/source/file/runner.go

```diff
@@ -51,23 +50,11 @@ func (r *runnerReader) Run(ctx context.Context) {
 		},
 	)
 
-	var wg sync.WaitGroup
-	wg.Add(1)
-
-	go func() {
-		defer wg.Done()
-		<-ctx.Done()
-		r.reader.Stop()
-	}()
-
 	for {
-		r.reader.Run()
+		r.reader.Run(ctx)
```

Comment:
This part is a bit problematic when using `decompressor` and potentially when using `tailer`. We will retry the until the target disappear. But when using `decompressor` we don't tail a file but read it fully until its finished. I wonder if we should quit them there then so we don't keep them around because its not necessary. 

I can think of a couple  of ways to do this, either set `MaxRetries` so we don't do this forever or return some kind of result from `Run` that can indicate if we should keep going or stop